### DISCUSSION TITLE
Help text for <ESC>

### DISCUSSION THREAD

**Discussion by ckuelker** [monkeytypegame/monkeytype#951] [Ideas]
Sorry, I am new to monkeytype and some technical suggestions might not be feasibly in CSS, HTML or Javascript.

When pressing <__ESC__> one gets a list of options to toggle. I guess this is called *"command line"*. This list is very handy and in many cases self explanatory. However sometimes I can not guess the meaning of an option. I used __1.7.3__ from https://monkeytype.com/

__Would it be possible to add a help text?__ For example when moving the mouse over the option (or a help symbol after the option)?

Actually there is a help text for each option under the `https://DOMAIN.TLD/settings` URL (I guess), that could be reused for
this purpose. I might be wrong here. 

Thank you for all your work!

**Comment by unrealapex** [comment]
What setting? Check if it is available on the settings page(https://monkeytype.com/settings). If it is there, there should be a brief description explaining it. If not I can explain the setting.

**Reply by ckuelker** [reply]
Thank you for answering and your contribution to monkey-type. The help text is great, I used it many times. 

I do not have a particular settings question at the moment. I stumbled over the somewhat cumbersome way to get to the help text, but it could be that I just do not understand a smart way of doing it.  If this is the case, please ignore the rest

I might not have been precise, so I will rephrase my point and answering your question. Regarding you question "What setting?":
1. enter https://monkeytype.com/
2. push ESC key
3. A intimidating list (A) appears in a pop up widget with setting names like:
 - some string...
 - difficult to remember string...
 - hard to understand string...
 - a very short string...

Maybe for you all this strings (setting options) are obvious. For a newbie it is not. And you are right, a possible way to understand is: go to  the help text on the settings page. So one would continue like this:

 4. remember that new string 
 5. go to https://monkeytype.com/settings page (B)
 6. scroll down the list until you find the string in a heading or inside the text
 7. read the help text section
 8. understand the meaning
 9. go back to page (A)
 10. change the setting
 
My suggestion to help to remember all this setting strings is to add a (mnemonic?) help text to (A) maybe with a direct link to (B). This would make it easier changing the settings on (A) and mostly staying at (A), without always the need to go to (B).  Like this

1. enter https://monkeytype.com/
2. push ESC key
3. A  better understandable list (C) appears in a pop up widget with setting names like:
 - some string... (Sets X)
 - difficult to remember string... (Modifies Y, but not Z)
 - hard to understand string... (Can not explain, see settings at W)
 - a very short string... (Alias of "some string", sets X)

Of course the solution also could be: just remember all the settings and meanings.

**Reply by unrealapex** [reply]
How about having the descriptions(in shortened form) appear in tooltips when a user hovers over an option in the command line for a few seconds?

**Reply by ckuelker** [reply]
This is a very good idea! For various reasons, it is superior to my suggestion.

**Reply by Miodec** [reply]
I dont think this is necessary. The settings page has in-depth descriptions of settings. The command line is meant to be used by users who already know what each setting does. Plus its meant to be used with your keyboard only (the mouse support is only there because people complained about it)
